src/json.ts

```diff
@@ -45,76 +45,20 @@ const finiteNumber = () =>
 
 /**
  * A struct to check if the given value is a valid JSON-serializable value.
- *
- * Note that this struct is unsafe. For safe validation, use {@link JsonStruct}.
  */
 // We cannot infer the type of the struct, because it is recursive.
-export const UnsafeJsonStruct: Struct<Json> = union([
+export const JsonStruct: Struct<Json> = union([
```

Comment:
I'm not sure if this is a good idea. These values should be considered "unsafe," since we don't do full validation on them. Renaming this may spark the idea that the value is completely validated, but in reality it's not.

Reply:
Ok, but it shouldn't be called `Json` then, as it's not fully safe either.

That's why I'm suggesting splitting structure validation, for which this is completely safe, and JS object sanitization, for which this is unsafe.

Reply:
New solution is added according to the review comments from this and https://github.com/MetaMask/snaps-monorepo/pull/1405.

Please note that I'm also trying to keep `isValidJson` working as well without having additional structures and having to change where it's used in the project since not all the places really need this type of sanitization for now (just the critical one identified on the second PR).